synapse/tests/test_cortex.py

```diff
@@ -2573,6 +2573,53 @@ class Bawk:
             self.len(2, nodes)
             self.eq(nodes[1].ndef, ('test:int', 25))
 
+    async def test_storm_filter_vars(self):
+        '''
+        Test variable filters (e.g. +$foo) and expression filters (e.g. +$(:hehe < 4))
+
+        '''
+        async with self.getTestCore() as core:
+
+            # variable filter, non-runtsafe, true path
+            q = '[test:type10=1 :strprop=1] $foo=:strprop +$foo'
+            nodes = await core.nodes(q)
+            self.len(1, nodes)
+
+            # variable filter, non-runtsafe, false path
+            q = '[test:type10=1 :strprop=1] $foo=:strprop -$foo'
+            nodes = await core.nodes(q)
+            self.len(0, nodes)
+
+            # variable filter, runtsafe, true path
```

Comment:
We should have a test where the filter is the form `#$var1=$var2` and `#$var1@=$var2` since tag filtering is a strong use here

Reply:
That functionality wasn't added in this PR...